src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -522,18 +574,33 @@ def distribute_shares(shares: List[ShareTensor], session: Session) -> List:
         Returns:
             List of ShareTensor.
 
+        Raises:
+            TypeError: when Datatype of shares is invalid.
+
         """
-        parties = session.parties
+        if not isinstance(shares, list):
+            raise TypeError("Shares to be distributed should be a list of shares")
 
+        if len(shares) == 0:
```

Comment:
MAYBE: Add an exception in the case where `len(shares)` is different than `nr_parties`